tools/slicec-cs/src/encoding.rs

```diff
@@ -2,61 +2,47 @@
 
 use crate::builders::{Builder, FunctionCallBuilder};
 use crate::cs_attributes::CsType;
+use crate::member_util::get_sorted_members;
 use crate::slicec_ext::*;
 use convert_case::Case;
 use slicec::code_block::CodeBlock;
 use slicec::grammar::*;
-use slicec::utils::code_gen_util::*;
+use slicec::utils::code_gen_util::{get_bit_sequence_size, TypeContext};
 
-pub fn encode_fields(fields: &[&Field], namespace: &str, encoding: Encoding) -> CodeBlock {
+pub fn encode_fields(fields: &[&Field], encoding: Encoding) -> CodeBlock {
     let mut code = CodeBlock::default();
 
-    let (required_fields, tagged_fields) = get_sorted_members(fields);
-
-    let bit_sequence_size = get_bit_sequence_size(encoding, &required_fields);
+    let bit_sequence_size = get_bit_sequence_size(encoding, fields);
     if bit_sequence_size > 0 {
         writeln!(
             code,
             "var bitSequenceWriter = encoder.GetBitSequenceWriter({bit_sequence_size});",
         );
     }
 
-    for field in required_fields {
-        let param = format!("this.{}", field.field_name());
-        code.writeln(&encode_type(
-            field.data_type(),
-            TypeContext::Field,
-            namespace,
-            &param,
-            "encoder",
-            encoding,
-        ));
-    }
-
-    // Encode tagged
-    for field in tagged_fields {
+    for field in get_sorted_members(fields) {
+        let namespace = field.namespace();
         let param = format!("this.{}", field.field_name());
-        code.writeln(&encode_tagged_type(
-            field,
-            namespace,
-            &param,
-            "encoder",
-            TypeContext::Field,
-            encoding,
-        ));
+        let encode_fn = match field.is_tagged() {
```

Comment:
Not sure if this is a rust convention, but to my eyes using an `if` for checking true/false seems better.

When I see a `match` I'm expecting more conditions, or something not covered by using plain `if`.